pkg/model/util.go

```diff
@@ -174,3 +174,13 @@ func roleMatches(a v1alpha1.RoleRepresentation, b v1alpha1.RoleRepresentation) b
 	}
 	return a.Name == b.Name
 }
+
+func filterEmptyImagePullSecrets(secrets []v1.LocalObjectReference) []v1.LocalObjectReference {
```

Comment:
Added this to make sure we aren't sending empty secrets during k8s API calls - which is probably okay to delegate to k8s server for validation but this is cleaner. Effectively, this PR won't modify the over-the-wire k8s API calls when the user doesn't specify any `imagePullSecret` environment variables.